src/client/client_impl.cc

```diff
@@ -79,6 +82,189 @@ google::protobuf::util::Status ClientImpl::CreateFile(
+    cache_file_chunk_metadata(filename, chunk_index, open_file_reply);
+  
+    chunk_handle = cache_manager_->GetChunkHandle(filename, chunk_index)
+                                       .ValueOrDie();
+    chunk_version = cache_manager_->GetChunkVersion(chunk_handle).ValueOrDie();
+    chunk_server_location_entry = 
+        cache_manager_->GetChunkServerLocation(chunk_handle).ValueOrDie();
+  }
+
+  // If no locations are there, we need to return error
+  if (chunk_server_location_entry.locations.empty()) {
+    return google::protobuf::util::Status(
+               google::protobuf::util::error::UNAVAILABLE,
+               "No chunk server has been found");
+  }
+
+  // Loop over all server locations but return immediately as long as one 
+  // read request is successful
+  for (auto& location : chunk_server_location_entry.locations) {
+    // Prepare for the read file chunk request
+    ReadFileChunkRequest read_file_chunk_request;
+    read_file_chunk_request.set_chunk_handle(chunk_handle);
+    read_file_chunk_request.set_chunk_version(chunk_version);
+    // We start reading from the beginning of this chunk, where the offset
+    // equals to chunk_index * common::bytes_per_mb
+    read_file_chunk_request.set_offset_start(chunk_index * 
+                                                 common::bytes_per_mb);
+    read_file_chunk_request.set_length(nbytes);
+    // Access the client-end-point for contacting the chunk server
+    auto server_address(location.server_hostname()+":" + 
+                            std::to_string(location.server_port()));
+    // Prepare the client context 
+    grpc::ClientContext client_context;  
+    common::SetClientContextDeadline(client_context, config_manager_); 
+    // Issue ReadFileChunkRequest and check status
+    StatusOr<ReadFileChunkReply> read_file_chunk_reply_or(
+        chunk_server_service_client_[server_address]->SendRequest(
+            read_file_chunk_request, client_context));
+    
+    // Handle grpc error, log and continue to try the next chunk server
+    if (!read_file_chunk_reply_or.ok()) {
+      LOG(ERROR) << "Read file chunk " << chunk_handle << " from "
+                 << server_address << " failed due to " 
+                 << read_file_chunk_reply_or.status().error_message();
+      continue;
+    }
+
+    // Handle chunk status error, log and continue
+    auto read_file_chunk_reply(read_file_chunk_reply_or.ValueOrDie());
+    switch (read_file_chunk_reply.status()) {
+      case ReadFileChunkReply::UNKNOWN:
+        LOG(ERROR) << "Unknown error while reading " + chunk_handle;
+        continue; 
+      case ReadFileChunkReply::FAILED_NOT_FOUND:
+        LOG(ERROR) << "Chunk not found: " + chunk_handle;
+        continue;
+      case ReadFileChunkReply::FAILED_STALE_VERSION:
+        LOG(ERROR) << "Reading stale version " << std::to_string(chunk_version)
+                   << " for chunk " + chunk_handle;
+        continue;
+      default: break;
```

Comment:
this default block doesn't seem necessary at all, since it gonna continue execute anyways

Reply:
I see a compiler warning without this so I added it.